lib/openid_connect/authorization_endpoint.rb

```diff
@@ -0,0 +1,82 @@
+class AuthorizationEndpoint
+  attr_accessor :app, :account, :client, :redirect_uri, :response_type, :scopes, :_request_, :request_uri, :request_object
+  delegate :call, to: :app
+
+  def initialize(allow_approval = false, approved = false)
+    @account = nil
+    @app = Rack::OAuth2::Server::Authorize.new do |req, res|
+      @client = nil # Find the client
+      res.redirect_uri = @redirect_uri = req.verify_redirect_uri!(@client.redirect_uris)
+      if res.protocol_params_location == :fragment && req.nonce.blank?
+        req.invalid_request! 'nonce required'
+      end
+      @scopes = req.scope.inject([]) do |_scopes_, scope|
+        _scopes_ << Scope.find_by_name(scope) or req.invalid_scope! "Unknown scope: #{scope}"
+      end
+      @request_object = if (@_request_ = req.request).present?
+                          OpenIDConnect::RequestObject.decode req.request, nil # @client.secret
+                        elsif (@request_uri = req.request_uri).present?
+                          OpenIDConnect::RequestObject.fetch req.request_uri, nil # @client.secret
+                        end
+      if Client.available_response_types.include? Array(req.response_type).collect(&:to_s).join(' ')
```

Comment:
Prefer `map` over `collect`.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.